### PR TITLE
closes #31 which adds a generic check for errors

### LINKED ISSUE #31: Better Error Checking
Currently the KeenAPI class will just check status code != 200 for errors.  When using the bulk add_events functionality the API will return 200 when malformed JSON is submitted (as well as for a number of other reasons).  The error messages are in the payload of the HTTP API.  This should be returning any errors in the payload or Keen's HTTP API should not be returning 200 (preferably both, as returning a 200 on errors isn't RESTful).

### PR DESCRIPTION
**Pull request by tushdante**
- modified the api.py to have a seperate function for error_handling
- added a check to find a generic error (by looking for the error_code in response)
- if error_code is found, it raises whatever error is in the response

### CODE REVIEW

**Comment by PeterNachbaur** [comment]
Thanks Tushar! Regarding #31 - I'll let the Middleware team know they should consider changing what code is returned for malformed JSON in a bulk event load.  

Can you also bump the version and update the changelog in the readme? Then I can show you how to get the new version out there.

**Comment by stroud109** [comment]
Thanks for working on this @tushdante! Could you change `if res.json().get("error_code"):`to `if response.status_code/100 != 2`? This way the code is agnostic of the content encoding.

**Comment by tushdante** [comment]
Sure thing, @stroud109 i'll update it and let you know :)

**Comment by tushdante** [comment]
Good catch @jdunck. I can do a `try catch` with the `res.json()` and raise and KeenApiError. If you think there's a better way, feel free to share. 

re: the `raise` regardless of status code, if you look at the code we're checking to see that the status code is like `2xx`. I'm doing that here ` if res.status_code/100 != 2:`

File: keen/api.py [hunk 1]
```diff
@@ -158,8 +154,17 @@ def query(self, analysis_type, params):
...
 
         return response.json()["result"]
+
+    def error_handling(self, res):
+        """
+        Helper function to do the error handling 
+
+        :params res: the response from a request
+        """
+        # making the error handling generic so if an status_code starting with 2 exists, we raise the error
```
Comment (jdunck) [inline keen/api.py]: This raises if it *isn't* 2xx (which is correct) but the comment is misleading.

**Comment by tushdante** [comment]
Oops, my bad. I'll fix the comment and figure out why the travis build is failing in the next commit.

**Comment by tushdante** [comment]
Hey @jdunck I'm having some trouble figuring out why my build fails with python 3 and above. Can you help me out?

**Comment by stroud109** [comment]
Looks good, @tushdante! Want to bump the version to 0.3.15?

**Comment by tushdante** [comment]
@stroud109 i've bumped up the version! i'd appreciate if you could just proofread the changes one last time (i may have messed something up while trying to merge the changes with the master branch). everything looks fine to me, though. thanks :)

File: README.md [hunk 2]
```diff
@@ -188,10 +188,16 @@ The Python client enables you to create [Scoped Keys](https://keen.io/docs/secur
...
 ### Changelog
 
+##### 0.3.15
++ Added better error handling to surface all errors from HTTP API calls 
+
 ##### 0.3.14
 + Added compatibility for pip 1.0
 
 ##### 0.3.13
++ Added better error handling to surface all errors from HTTP API calls
```
Comment (jdunck) [inline README.md]: 0.3.13 is repeated here, probably you meant to take out these lines and just list it as 0.3.15?

File: keen/api.py [hunk 3]
```diff
@@ -158,8 +154,17 @@ def query(self, analysis_type, params):
...
+
+    def error_handling(self, res):
+        """
+        Helper function to do the error handling 
+
+        :params res: the response from a request
+        """
+        # making the error handling generic so if an status_code starting with 2 doesn't exist, we raise the error
+        if res.status_code // 100 != 2:
+            error = res.json()
```
Comment (jdunck) [inline keen/api.py]: You forgot to handle the non-JSON case.  How about this?

```
from requests.compat import json

try:
    error = res.json()
except json.JSONDecodeError:
    error = {'message': 'The API did not respond with JSON, but: "{0}"'.format(res.text[:1000])', "error_code": "InvalidResponseFormat"}
```

**Comment by tushdante** [comment]
Thanks @jdunck ! I updated the changes, I'd love your feed back for the same. Especially with `keen/api.py`

**Comment by jdunck** [comment]
:+1:

**Pull request merged**